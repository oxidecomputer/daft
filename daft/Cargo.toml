[package]
name = "daft"
version = "0.1.0"
description = "Structural diffs of Rust data structures"
edition.workspace = true
repository.workspace = true
rust-version.workspace = true
license.workspace = true

[lints]
workspace = true

[dependencies]
daft-derive.workspace = true
newtype-uuid = { workspace = true, optional = true }
oxnet = { workspace = true, optional = true }
paste.workspace = true
uuid = { workspace = true, optional = true, features = ["v4"] }

[features]
newtype-uuid1 = ["dep:newtype-uuid"]
oxnet01 = ["dep:oxnet"]
uuid1 = ["dep:uuid"]
