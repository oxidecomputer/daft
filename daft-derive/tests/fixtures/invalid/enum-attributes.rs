use daft::Diffable;

#[derive(Diffable)]
#[daft(leaf)]
enum MyEnum {
    #[daft(leaf)]
    A(#[daft(ignore)] i32),
    #[daft(leaf)]
    B {
        #[daft(ignore)]
        data: usize,
    },
}

#[derive(Diffable)]
#[daft(ignore)]
enum MyEnum2 {
    A(i32),
}

fn main() {
    // MyEnum/MyEnum2 and its Diffable impl should all exist.
    let before = MyEnum::A(0);
    let after = MyEnum::B { data: 1 };

    let _diff = before.diff(&after);

    let before = MyEnum2::A(0);
    let after = MyEnum2::A(1);

    let _diff = before.diff(&after);
}
