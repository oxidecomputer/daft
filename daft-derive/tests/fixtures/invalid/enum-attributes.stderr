error: this is unnecessary: the Diffable implementation for enums is always a leaf
 --> tests/fixtures/invalid/enum-attributes.rs:4:8
  |
4 | #[daft(leaf)]
  |        ^^^^

error: daft attributes are not allowed on enum variants
 --> tests/fixtures/invalid/enum-attributes.rs:6:5
  |
6 |     #[daft(leaf)]
  |     ^^^^^^^^^^^^^

error: daft attributes are not allowed on enum variant fields
 --> tests/fixtures/invalid/enum-attributes.rs:7:7
  |
7 |     A(#[daft(ignore)] i32),
  |       ^^^^^^^^^^^^^^^

error: daft attributes are not allowed on enum variants
 --> tests/fixtures/invalid/enum-attributes.rs:8:5
  |
8 |     #[daft(leaf)]
  |     ^^^^^^^^^^^^^

error: daft attributes are not allowed on enum variant fields
  --> tests/fixtures/invalid/enum-attributes.rs:10:9
   |
10 |         #[daft(ignore)]
   |         ^^^^^^^^^^^^^^^

error: daft attributes are not allowed on enums
  --> tests/fixtures/invalid/enum-attributes.rs:16:8
   |
16 | #[daft(ignore)]
   |        ^^^^^^
